smt/pkg/smt/smt_state_writer.go

```diff
@@ -0,0 +1,140 @@
+package smt
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+
+	"github.com/holiman/uint256"
+	"github.com/ledgerwatch/erigon-lib/common"
+	libcommon "github.com/ledgerwatch/erigon-lib/common"
+	"github.com/ledgerwatch/erigon/core/state"
+	"github.com/ledgerwatch/erigon/core/types"
+	"github.com/ledgerwatch/erigon/core/types/accounts"
+	"github.com/ledgerwatch/erigon/smt/pkg/utils"
+)
+
+var _ state.StateWriter = (*SMT)(nil)
+
+func (s *SMT) UpdateAccountData(address common.Address, _ *accounts.Account, account *accounts.Account) error {
+	if err := s.SetAccountStorage(address, account); err != nil {
+		return err
+	}
+
+	// TODO: @Stefan-Ethernal How to update:
+	// 1. the account bytecode (we have CodeHash, but not the Code itself, so how to retrieve the actual Code) and
+	// 2. the storage slots (we only have the Root, but not the storage slots themselves)
```

Comment:
We don't need to update code or storage slot, which will be updated in other functions below.